### ISSUE TITLE
bashism

### ISSUE THREAD

**Issue by duck-rh** [CISOfy/lynis#1269]
**Describe the bug**
A few bashism were detected by `checkbashisms` (a tool in Debian `devscripts` package) and reported by lintian. Many distributions do not use bash as/bin/sh and that could cause problems (freebsd, debian uses dash…)

**Version**
 - Distribution Debian unstable
 - Lynis version 3.0.7

**Expected behavior**
bashisms replaced by POSIX equivalents.

**Output**
Here is the list of detected problems, excluding the false-positive on HOSTNAME:
```
I: lynis: bash-term-in-posix-shell '. ${INCLUDEDIR}/helper_${HELPER} ${HELPER_PARAMS}' [usr/sbin/lynis:1095]
I: lynis: bash-term-in-posix-shell '[ "$(uname -o 2> /dev/null)" ==' [usr/share/lynis/include/osdetection:681]
I: lynis: bash-term-in-posix-shell 'typeset' [usr/share/lynis/include/functions:2534]
```

The problems being, in order, use of dot instead of `source`, use of `==` and use of 'typeset`.

**Comment by mboelen** [comment]
Thanks for sharing!

First item: dot is not a bashism? Looking at some of the [shell POSIX specification](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html). To find the related section search for "dot - execute commands in the current environment". The usage of `.` was there since the very beginning of Lynis (14 years ago) and never caused any issues that I'm aware of.

Second item: good catch, that one is fixed now with commit https://github.com/CISOfy/lynis/commit/66f93f0275a8188537934186ee5a44e5c0262f75

Third item: yes, `typeset` is normally not to be used. In this case, it's only used when it is available on the system.

**Comment by duck-rh** [comment]
@mboelen thanks for looking into it.

Indeed for the dot/source that's the opposite, I was mistaken. I looked into it and the problem is the use of parameters. I tested it on my machine where sh is dash and indeed `$*` contains parameters passed to the script and not to the sourced file. It seems that `HELPER_PARAMS` actually does not work but since you `shift` parameters it is not needed and helpers get what's needed into `$#` and `$1` etc.
